\documentclass[../main.tex]{subfiles}
\graphicspath{{\subfix{../figures/}}}

\begin{document}
\begin{abstract}

{\parindent0pt Experiments investigating the regulation of RNA transcripts have been revolutionised by technology developed over the last 40 years.
The data acquired from these experiments have revealed novel regulatory mechanisms for the localisation, degradation and modification of RNA transcripts.
However, the volume and complexity of the data sets has led to an unprecedented reliance on statistical software.
Inadequate analysis of data sets is contributing to the ongoing crisis around reproducing conclusions from published research.
Rigorous implementation of statistical analysis software can continue to uncover novel regulatory mechanisms, but closed, obscure, and incorrect analyses will propagate the reproducibility crisis to unassailable new heights.}

The objective of this research project is to develop open source software and implement reproducible analyses to enable further exploration of regulatory mechanisms acting on RNA transcripts. 
This thesis focuses on the analysis of transcriptomics data sets, predominately from the model organism \textit{Saccharomyces cerevisiae}.
This first project discusses the standardisation of the analysis of qPCR data. 
The chapter compares the R package tidyqpcr, developed by the author, to other current software available. 
This case highlights how quality software supported by comprehensive documentation can improve the quality of an entire experimental assay.
The next chapter showcases how the implementation of quality analysis can detect subtle interactions between regulatory motifs.
The design of several reporter constructs using insights from published data sets shows how even short regulatory motifs can be affected by their overall context.
The final results chapter outlines the development of a statistical software package to rigorously analyse noisy transcriptomic data from RNA-Seq assays exploring RNA localisation.
The statistical software package uses a Bayesian hierarchical model of fractionation-based assays to overcome common biases in RNA-Seq data sets.

In summary, this thesis presents and implements two examples of research software that improve the reproducibility and quality of conclusions from data acquired from common experimental assays in molecular biology. 
The thesis also outlines how to implement open source development practices and create inclusive documentation in an academic setting. 
Software developed within this framework is then used to elucidate subtle ways that cells regulate their transcriptome.

\end{abstract}
\end{document}