\documentclass[../main.tex]{subfiles}
\graphicspath{{\subfix{../figures/}}}

\begin{document}
\begin{abstract}

Molecular biology has entered an age of unprecedented volumes of data.
The growth of available data has overwhelmed manual curation and has led to a dependence on computational methods to automate analysis.
Yet, the complexity of regulatory mechanisms within cells still dwarfs the available information. 
The perilous situation of having too much data to inspect manually, but too little high quality data to describe a process in its entirety is a significant challenge to modern research.
Inadequate analysis is contributing to the ongoing crisis around reproducing conclusions from published research.
Rigorous implementations of statistical analysis software can uncover phenomena at incredible scales, but closed, obscure, and incorrect analyses can propagate the reproducibility crisis to unassailable new heights.

The objective of this research project is the implementation of open source software practices and the pedagogy of software documentation to promote rigorous statistical analysis through the development of quality research software. 
The thesis focuses on the analysis of transcriptomics data sets, predominately in the model organism \textit{Saccharomyces} cerevisiae.
The first project discusses the issue of the standardisation of qPCR data analysis. 
The chapter contrasts the R package tidyqpcr, developed by the author, to other current software available. 
This use case highlights how quality software supported by comprehensive documentation can improve the quality of an entire experimental assay.
The next chapter showcases how the implementation of quality analysis can detect subtle interactions between regulatory motifs.
The careful design of several reporter constructs using insights from already published data sets shows how even short regulatory motifs can be affected by their overall context.
The final results chapter outlines the development of a statistical software package to rigorously analyse noisy transcriptomic data from a non-standard RNA-Seq assay.
Several software packages have been developed to counter the significant biases present in RNA-Seq data sets.
However, their generality requires them to make assumptions about the overall structure of the RNA data sets; assumptions that are broken in many RNA-seq assays.
The versatility of Bayesian hierarchical models at incorporating complex error structures is used to create software to analysis RNA-Seq assays exploring the localisation of mRNA transcripts through differential fractionation.

In summary, this thesis presents and implements two examples of research software that improve the reproducibilty of and the quality of conclusions from data acquired from common experimental assays in molecular biology. 
The thesis also outlines how to implement open source development practices and create inclusive documentation in an academic setting. 
Software developed within this framework is then used to elucidate subtle ways that cells regulate their transcriptome.

\end{abstract}
\end{document}